public/app/features/alerting/unified/components/receivers/TemplatePreview.tsx

```diff
@@ -38,23 +33,14 @@ export function TemplatePreview({
 
   const templateContent = watch('content');
 
-  const [trigger, { data, error: previewError, isLoading }] = usePreviewTemplateMutation();
-
+  const {
+    data,
+    isLoading,
+    onPreview,
+    error: previewError,
+  } = usePreviewTemplate(templateContent, templateName, payload, setPayloadFormatError);
   const previewToRender = getPreviewResults(previewError, payloadFormatError, data);
 
-  const onPreview = useCallback(() => {
```

Comment:
moved to a separate hook, so we can re use it in the `TemplateContentAndPreview` component